generators/Reporting.fs

```diff
@@ -0,0 +1,92 @@
+    | Some Status.All,           _                        -> true
+    | _ -> false
+    
+type SummaryTypes =
+    | Custom of string
+    | Unimplemented of string
+    | MissingData of string
+    | Deprecated of string
+    | UpToDate of string
+    | Outdated of string * string * string
+    
+let nameFromSummaryType = function
+    | Custom n -> n
+    | Unimplemented  n -> n
+    | MissingData n -> n
+    | Deprecated n -> n
+    | UpToDate n -> n
+    | Outdated (n,_,_) -> n
+    
+let private summarizeExercise options (parseCanonicalData':string -> CanonicalData) (exercise:Exercise)  =
+
+    match exercise with
+    | Exercise.Custom custom ->                 SummaryTypes.Custom custom.Name
+    | Exercise.Unimplemented unimplemented ->   SummaryTypes.Unimplemented unimplemented.Name
+    | Exercise.MissingData missingData ->       SummaryTypes.MissingData missingData.Name
+    | Exercise.Deprecated deprecated ->         SummaryTypes.Deprecated deprecated.Name
+    | Exercise.Generator generator ->
+        let cData = parseCanonicalData' generator.Name
+        
+        match generator.ReadVersion (), cData.Version with
+        | a,b when a.Equals b ->  UpToDate generator.Name
+        | a,b -> Outdated (generator.Name,a,b)
+        
+let listExercises options =
+    Log.Information(sprintf "Listing exercises with status %s..." (options.Status.Value.ToString()))
+    
+    let parseCanonicalData' = CanonicalData.parseCanonicalData options
+    
+    let allExercises = createExercises options
+    
+    let longestNameLength = allExercises |> List.map exerciseName |> List.map (fun e -> e.Length) |> List.max
+    let indentAfter (name:string) = String.replicate (longestNameLength+2-name.Length) " "
+    
+    let exerciseGroups = allExercises
+                            |> List.filter (filterByStatus options parseCanonicalData')
+                            |> List.map (summarizeExercise options parseCanonicalData')
+                            |> List.groupBy (fun x -> x.GetType())
+    
+    exerciseGroups |> List.iter (fun (groupType,group) ->
+        let description = match group.Head with
+                          | SummaryTypes.Custom _ -> "have customized tests"
+                          | SummaryTypes.Unimplemented _ -> "are missing test generator"
+                          | SummaryTypes.MissingData _ -> "are missing canonical data"
+                          | SummaryTypes.Deprecated _ -> "are deprecated"
+                          | SummaryTypes.UpToDate _ -> "are up to date"
+                          | SummaryTypes.Outdated _ -> "are outdated"
+                          
+        Log.Information("{num} exercises {description}:", group.Length, description)
+        
+        group |> List.iter (fun exercise -> 
+            match exercise with
+            | SummaryTypes.Outdated (name,oldVer,newVer) -> Log.Information(" {name}{indent}({oldVer} -> {newVer})", name, indentAfter name, oldVer, newVer)
```

Comment:
I was thinking that maybe `Log.Warning` would be a better logging level perhaps for outdated exercises? What do you think?

Reply:
I'm not in the habit of using logging libraries for small command-line tools like this. What I think is that we could remove the first column of output entirely ;)

If this were a warning, then should some of the other categories be warnings too?

Reply:
Once I started using `Log.Information("{foo} {bar}", foo, bar)` instead of `Log.Information(sprintf "%s %s" foo bar)`, I came to really like the colors, too.

Maybe we could configure Serilog to leave the first column off, but keep the benefit of colorizing the output?

Reply:
Sounds like a plan! If you wish to include that in this PR, that would be fine. A separate PR would be better :)